From 7c66d5a3f396a74c15346b908bd8a2a3527bc6f6 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:13 -0400
Subject: [infiniband] qib: Remove ib_sg_dma_address() and ib_sg_dma_len() overloads

Message-id: <195067e484e8bee15ffa32d7cdddaefe4c30cb42.1410425016.git.avadai@redhat.com>
Patchwork-id: 93579
O-Subject: [Patch RHEL7.1 V1 017/197] IB/qib: Remove ib_sg_dma_address() and ib_sg_dma_len() overloads
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Mike Marciniszyn <mike.marciniszyn@intel.com>

Remove the overload for .dma_len and .dma_address

The removal of these methods is compensated for by code changes to
.map_sg to insure that the vanilla sg_dma_address() and sg_dma_len()
will do the same thing as the equivalent former ib_sg_dma_address()
and ib_sg_dma_len() calls into the drivers.

Suggested-by: Bart Van Assche <bvanassche@acm.org>
Reviewed-by: Dennis Dalessandro <dennis.dalessandro@intel.com>
Tested-by: Vinod Kumar <vinod.kumar@intel.com>
Signed-off-by: Mike Marciniszyn <mike.marciniszyn@intel.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 446bf432a9b084d9f3471eca309cc53fa434ccc7)
Signed-off-by: Amir Vadai <amirv@mellanox.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/qib/qib_dma.c b/drivers/infiniband/hw/qib/qib_dma.c
index 2920bb3..59fe092 100644
--- a/drivers/infiniband/hw/qib/qib_dma.c
+++ b/drivers/infiniband/hw/qib/qib_dma.c
@@ -108,6 +108,10 @@ static int qib_map_sg(struct ib_device *dev, struct scatterlist *sgl,
    ret = 0;
    break;
   }
+  sg->dma_address = addr + sg->offset;
+#ifdef CONFIG_NEED_SG_DMA_LENGTH
+  sg->dma_length = sg->length;
+#endif
  }
  return ret;
 }
@@ -119,21 +123,6 @@ static void qib_unmap_sg(struct ib_device *dev,
  BUG_ON(!valid_dma_direction(direction));
 }
 
-static u64 qib_sg_dma_address(struct ib_device *dev, struct scatterlist *sg)
-{
- u64 addr = (u64) page_address(sg_page(sg));
-
- if (addr)
-  addr += sg->offset;
- return addr;
-}
-
-static unsigned int qib_sg_dma_len(struct ib_device *dev,
-       struct scatterlist *sg)
-{
- return sg->length;
-}
-
 static void qib_sync_single_for_cpu(struct ib_device *dev, u64 addr,
         size_t size, enum dma_data_direction dir)
 {
@@ -173,8 +162,6 @@ struct ib_dma_mapping_ops qib_dma_mapping_ops = {
  .unmap_page = qib_dma_unmap_page,
  .map_sg = qib_map_sg,
  .unmap_sg = qib_unmap_sg,
- .dma_address = qib_sg_dma_address,
- .dma_len = qib_sg_dma_len,
  .sync_single_for_cpu = qib_sync_single_for_cpu,
  .sync_single_for_device = qib_sync_single_for_device,
  .alloc_coherent = qib_dma_alloc_coherent,
-- 
1.7.1